### DISCUSSION TITLE
graphql.schema.validation.InvalidSchemaException: invalid schema not implementing interface

### DISCUSSION THREAD

**Discussion by drohne1673** [graphql-java/graphql-java#3701] [Q&A] [answered]
i was running WildFly 30 → Feature pack 2.2.0.Final →SmallRye GraphQL 2.5.0 included. -> graphql-java 20.2
i now upgraded to WildFly 33 →Feature pack 2.4.1.Final →SmallRye GraphQL 2.9.1 included -> graphql-java 22.1

now i get these errors when starting up:

> Caused by: graphql.schema.validation.InvalidSchemaException: invalid schema:
> object type 'BasicOrder' does not implement interface 'Order' because field 'orderId' is defined as 'String' type and not as 'String!' type
> object type 'BasicOrder' does not implement interface 'Order' because field 'orderState' is defined as 'OrderState' type and not as 'OrderState!' type"}}

the classes are quite simple:

```
public interface Order {
	default String getOrderType() {
		return getClass().getSimpleName();
	}

	@NonNull
	String getOrderId();

	@NonNull
	OrderState getOrderState();
}
```

and

```
@Description("A generic order for basic orderInformation to return to the client")
@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
public class BasicOrder implements Order {

	@NonNull
	@Setter(NONE) // not in GraphQL Input
	private String orderId;

	@NonNull
	@Setter(NONE) // not in GraphQL Input
	private OrderState orderState;
}
```

has aynone an idea why this may cause graphql trouble ? thanks

**Comment by bbakerman** [comment, marked as answer]
I am not sure what WF33.0 but I am assuming its an application framework - I know SmallRye is an frameowrk over top of the graphql-java base library.

I am gathering these frameworks convert Java declarations into graphql types dynamically at runtime (or many build time).

I can say the error messages tells me that the interface most likely got generated as

```
interface Order {
   orderId : String!
   orderState : OrderState!
}
```

and that BasicOrder got made as

```
type BasicOrder implements Order {
   orderId : String
   orderState : OrderState
}
```

eg the non null ness did not get applied for some reason.  But I don't know WHY these frameworks did this or how the generate their graphql types from Java declarations of code.

Perhaps you need to report it to the WF project (whatever that is) or SmallRye project maybe ?

**Reply by bbakerman** [reply]
WF is this Wild Fly ??

**Reply by drohne1673** [reply]
Yes its Wildfly, installing graphql-java via https://github.com/wildfly-extras/wildfly-graphql-feature-pack

**Reply by drohne1673** [reply]
so i created an issue there https://github.com/wildfly-extras/wildfly-graphql-feature-pack/issues/180

**Reply by phillip-kruger** [reply]
Probably best to report this in SmallRye GraphQL: https://github.com/smallrye/smallrye-graphql

**Reply by drohne1673** [reply]
@phillip-kruger in the end its the same people :) and as i wrote i already opened the feature-pack issue
they already proposed an idea. so thanks. i mark is as answered
